### PR TITLE
Use CopyableText in devtools

### PR DESCRIPTION
**Pull request by gsouquet**
## Checklist

-   [x] Tests written for new code (and old code if feasible)
-   [x] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

File: test/components/views/dialogs/DevtoolsDialog-test.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    it("copies the roomid", async () => {
+        const user = userEvent.setup();
+        const { container } = getComponent(room.roomId);
+
+        const copyBtn = getByLabelText(container, "Copy");
+        user.click(copyBtn);
+
+        // Not great, but I could not find another way to wait for the clipboard
+        // to be populated
+        // Ideally would prefer to write `waitFor(() => expect(navigator.clipboard.write).toHaveBeenCalled())
```
Comment (alunturner) [inline test/components/views/dialogs/DevtoolsDialog-test.tsx:63]: This test appears to be failing (at least according to what I can see on Github, so I wonder if this idea was good, and it was the next test that was failing?

Alternatively I wonder in this case if you could turn `navigator.clipboard` into a mock fn by spying on it, then check to see that it has been called and what it has been called with?

Comment (gsouquet) [reply test/components/views/dialogs/DevtoolsDialog-test.tsx:63]: `userEvent.setup()` is supposed to do just that... https://testing-library.com/docs/user-event/clipboard/

But it does not seem to work here, I'm failing to find much information online

File: src/components/views/dialogs/DevtoolsDialog.tsx [hunk 2]
```diff
@@ -119,11 +120,15 @@ const DevtoolsDialog: React.FC<IProps> = ({ roomId, onFinished }) => {
...
+                        </CopyableText>
                         <div className="mx_DevTools_label_bottom" />
-                        <DevtoolsContext.Provider value={{ room: cli.getRoom(roomId) }}>
-                            {body}
-                        </DevtoolsContext.Provider>
+                        {cli.getRoom(roomId) && (
+                            <DevtoolsContext.Provider value={{ room: cli.getRoom(roomId)! }}>
+                                {body}
+                            </DevtoolsContext.Provider>
+                        )}
```
Comment (t3chguy) [inline src/components/views/dialogs/DevtoolsDialog.tsx:131]: this change looks unrelated to the PR?

Comment (gsouquet) [reply src/components/views/dialogs/DevtoolsDialog.tsx:131]: Very true, it was there to appease the TypeStrict checks

**Review by t3chguy — APPROVED** [review]
